src/App.jsx

```diff
@@ -14,33 +16,93 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const ALPHABIT = 'alphabetically';
+const LENGTH = 'by length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedList(goods, reverse, { sortField }) {
+  const preparedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case ALPHABIT:
+          return good1.localeCompare(good2);
+        case LENGTH:
+          return good1.length - good2.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (reverse) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReversed, setIsReversed] = useState(false);
+  const goods = getSortedList(goodsFromServer, isReversed, { sortField });
```

Comment:
Not critical for now - but you repeat this on each component render

The easiest way to fix is 
1. useEffect + useState
2. useMemo